src/main/kotlin/com/hj/leetcode/kotlin/problem1046/Solution.kt

```diff
@@ -0,0 +1,36 @@
+package com.hj.leetcode.kotlin.problem1046
+
+import java.util.*
+
+/**
+ * LeetCode page: [1046. Last Stone Weight](https://leetcode.com/problems/last-stone-weight/);
+ */
+class Solution {
+    /* Complexity:
+     * Time O(NLogN) and Space O(N) where N is the size of stones;
+     */
+    fun lastStoneWeight(stones: IntArray): Int {
+        val stoneMaxPq = PriorityQueue<Int>(reverseOrder()).apply {
+            offerStones(stones)
+            playUntilEndOfGame()
+        }
+        val lastStone = stoneMaxPq.firstOrNull()
+        return lastStone ?: 0
+    }
+
+    private fun PriorityQueue<Int>.offerStones(stones: IntArray) {
```

Comment:
The context of function is unclear

Reply:
Refactor at cb4fc268512c82e9e09168f08512322a72c3aa64 and 2f2c88d63e84ecce336e39167539c26a9118a170